server/conditions/conditions.go

```diff
@@ -1,15 +1,21 @@
 package conditions
 
 import (
+	"fmt"
 	"github.com/bitdecaygames/fireport/server/actions"
 	"github.com/bitdecaygames/fireport/server/cards"
 	"github.com/bitdecaygames/fireport/server/pogo"
 	"github.com/pkg/errors"
 )
 
+const (
+	// MaxConditionSteps the maximum number of steps a condition will make before throwing an error
+	MaxConditionSteps = 10
+)
+
 // Condition checks each ActionGroup for a specific condition and modifies that ActionGroup if necessary
 type Condition interface {
-	Apply(gameState *pogo.GameState, actionGroup []actions.Action) error
+	Apply(gameState *pogo.GameState, actionGroup []actions.Action, step int) (int, error)
```

Comment:
Added a way to track how many steps the given condition has taken.  So you start off by calling this with 0, and it will return with the number of steps the condition took to complete.